core/src/main/java/discord4j/core/spec/WebhookExecuteSpec.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * This file is part of Discord4J.
+ *
+ * Discord4J is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Discord4J is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Discord4J. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package discord4j.core.spec;
+
+import discord4j.core.object.entity.Attachment;
+import discord4j.core.object.entity.channel.AllowedMentions;
+import discord4j.discordjson.json.AllowedMentionsData;
+import discord4j.discordjson.json.EmbedData;
+import discord4j.discordjson.json.WebhookExecuteRequest;
+import discord4j.discordjson.possible.Possible;
+import discord4j.rest.util.MultipartRequest;
+import reactor.util.function.Tuple2;
+import reactor.util.function.Tuples;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Spec used to create {@link discord4j.core.object.entity.Message Messages} via a {@link discord4j.core.object.entity.Webhook}.
+ *
+ * @see <a href="https://discord.com/developers/docs/resources/webhook#execute-webhook">Execute webhook</a>
+ */
+public class WebhookExecuteSpec implements Spec<MultipartRequest> {
```

Comment:
I wish there could be some code sharing here with MessageCreateSpec, but it's probably not worth it. Is the only difference that webhooks can send multiple embeds?

Reply:
The webhook can also specify the name and avatar of the messages.

Reply:
Let's leave any code sharing improvement to later spec changes.